addons/website/models/website_menu.py

```diff
@@ -62,6 +62,28 @@ def name_get(self):
             res.append((menu.id, menu_name))
         return res
 
+    def _validate_parent_menu(self):
+        for record in self:
+            parent_menu = record.parent_id.sudo() if record.parent_id else None
+
+            if parent_menu:
+                # Check hierarchy level
+                if parent_menu.parent_id and parent_menu.parent_id.parent_id:
```

Comment:
Why we need ```parent_menu.parent_id.parent_id``` I guess ```parent_menu.parent_id```

Reply:
We need to ensure that the parent menu has a parent_id. If it does, we should then check its parent_id.
For example, consider the following menu hierarchy:
1. Top menu -> parent_id: null
2. Main -> parent_id: 1
3 .Child1 -> parent_id: 2 (This menu should be allowed).
If we remove this check, it will not allow the creation of such a menu hierarchy.
https://www.loom.com/share/5a1496f38f0443b5b4f1bcab7e665ac8?sid=27f3ec4c-3943-4491-a983-a1bdcc55fce0